From 5cfafd07096628d78a685e13f2e765f22ac44c13 Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@aries-embedded.de>
Date: Tue, 10 Nov 2020 10:58:34 +0100
Subject: [PATCH 08/11] ARM: dts: imx28-m28evk: Convert to the new display
 bindings

imx28-evk board uses a Ampire AM-800480R2 800x480 7" panel display.

Instead of hardcoding the display timings in the device tree, use
the "ampire,am800480r2" compatible instead.

This aligns with the new mxsfb bindings scheme documented at:
Documentation/devicetree/bindings/display/mxsfb.txt

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 arch/arm/boot/dts/imx28-m28evk.dts | 48 +++++++++++++++---------------
 1 file changed, 24 insertions(+), 24 deletions(-)

diff --git a/arch/arm/boot/dts/imx28-m28evk.dts b/arch/arm/boot/dts/imx28-m28evk.dts
index c870891e293b..7c10c329e328 100644
--- a/arch/arm/boot/dts/imx28-m28evk.dts
+++ b/arch/arm/boot/dts/imx28-m28evk.dts
@@ -11,6 +11,19 @@ / {
 	model = "Aries M28EVK";
 	compatible = "aries,m28evk", "fsl,imx28";
 
+	panel {
+		compatible = "ampire,am800480r2";
+		backlight = <&backlight_display>;
+		dvdd-supply = <&reg_3p3v>;
+		avdd-supply = <&reg_lcd_5v>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&display_out>;
+			};
+		};
+	};
+
 	apb@80000000 {
 		apbh@80000000 {
 			ssp0: spi@80010000 {
@@ -76,30 +89,10 @@ lcdif@80030000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&lcdif_24bit_pins_a
 					     &lcdif_pins_m28>;
-				display = <&display0>;
 				status = "okay";
-
-				display0: display0 {
-					bits-per-pixel = <16>;
-					bus-width = <18>;
-
-					display-timings {
-						native-mode = <&timing0>;
-						timing0: timing0 {
-							clock-frequency = <33260000>;
-							hactive = <800>;
-							vactive = <480>;
-							hback-porch = <0>;
-							hfront-porch = <256>;
-							vback-porch = <0>;
-							vfront-porch = <45>;
-							hsync-len = <1>;
-							vsync-len = <1>;
-							hsync-active = <0>;
-							vsync-active = <0>;
-							de-active = <1>;
-							pixelclk-active = <1>;
-						};
+				port {
+					display_out: endpoint {
+						remote-endpoint = <&panel_in>;
 					};
 				};
 			};
@@ -225,7 +218,7 @@ mac1: ethernet@800f4000 {
 		};
 	};
 
-	backlight {
+	backlight_display: backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm 4 5000000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
@@ -258,6 +251,13 @@ reg_usb1_vbus: regulator@3 {
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
 			gpio = <&gpio3 13 0>;
+	        };
+
+		reg_lcd_5v: regulator@4 {
+			compatible = "regulator-fixed";
+			regulator-name = "lcd-5v";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
 		};
 	};
 
-- 
2.34.1

